Userland/Libraries/LibC/tls.cpp

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2023, Daniel Bertalan <dani@danielbertalan.dev>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Types.h>
+#include <sys/internals.h>
+
+extern "C" {
+dtv_list_node __dtv_list;
+
+void* __tls_get_addr(__tls_index* index)
+{
+    size_t module_id = index->ti_module;
+    size_t module_offset = 0;
+    for (auto* node = &__dtv_list; node; node = node->next) {
+        if (module_id < dtv_list_node::capacity) [[likely]] {
+            module_offset = node->dtv[module_id];
+            break;
+        }
+        module_id -= dtv_list_node::capacity;
+    }
```

Comment:
As we only support static TLS, all threads share a single `dtv` array (actually, a chunked [linked list](https://discord.com/channels/830522505605283862/864504515818029087/1125868585823256637) to be able to append to it without creating races) which stores the thread-pointer offsets of each module's thread-local variables within the static TLS block.